mantidproject/mantid
# Add a checkbox to freeze the rotation

**Description of work.**


Add a checkbox to freeze the rotation in the instrument view, when in Full3D. This is useful for the feature added in #30343 , where the user can draw sectors for integration, but those sectors must be drawn with the viewport in the z = 0 plane, which can be difficult to achieve for mac users.
Thus checking this box makes left click behave like right click, ie manage the translation.

**To test:**
- Load a file, open the instrument view, and check that when the checkbox is unticked, it behaves as usual, whereas when it is ticked, the left-click translate the view.
- Save to project with the view opened and the checkbox ticked, and check the checkbox is correctly ticked when the project is loaded.


Fixes #30481 .






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
